src/components/inputs/time-input/time-input.js

```diff
@@ -23,144 +23,133 @@ const format24hr = ({ hours, minutes, seconds, milliseconds }) => {
 
 const hasMilliseconds = parsedTime => parsedTime.milliseconds !== 0;
 
-export class TimeInput extends React.Component {
-  static displayName = 'TimeInput';
-
-  // Takes any input like 15:10, 3 AM, 3AM, 3:15AM, 3:5AM and turns it
-  // into a 24h format (with seconds and milliseconds if present)
-  static to24h = time => {
-    const parsedTime = parseTime(time);
-    return parsedTime ? format24hr(parsedTime) : '';
-  };
-
-  static propTypes = {
-    id: PropTypes.string,
-    name: PropTypes.string,
-    autoComplete: PropTypes.string,
-    value: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func,
-    hasError: PropTypes.bool,
-    isAutofocussed: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    isReadOnly: PropTypes.bool,
-    placeholder: PropTypes.string,
-    horizontalConstraint: PropTypes.oneOf(['s', 'm', 'l', 'xl', 'scale']),
-
-    // HoC
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-      formatTime: PropTypes.func.isRequired,
-      locale: PropTypes.string.isRequired,
-    }).isRequired,
-  };
-
-  state = {
-    // We generate an id in case no id is provided by the parent to attach the
-    // label to the input component.
-    id: this.props.id,
-  };
-
-  static getDerivedStateFromProps = (props, state) => ({
-    id: getFieldId(props, state, sequentialId),
-  });
-
-  static defaultProps = {
-    horizontalConstraint: 'scale',
-  };
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.intl.locale !== this.props.intl.locale) {
-      this.emitChange(
-        TimeInput.toLocaleTime(this.props.value, this.props.intl.locale)
-      );
+const TimeInput = props => {
+  const id = getFieldId(props, {}, sequentialId);
+  const intl = useIntl();
+
+  const emitChange = React.useCallback(
+    value => {
+      const event = {
+        target: { id, name: props.name, value },
+      };
+      props.onChange(event);
+    },
+    [id, props.name]
```

Comment:
Is not onChange also a dep?

Reply:
Do you think it's relevant to put `onChange` as a dependency? When would the function ever change? 🤔

Reply:
Yeah, I noted that on another PR.

Reply:
Link?

Reply:
You reacted to it darling :) https://github.com/commercetools/ui-kit/pull/958#discussion_r307601310

Reply:
Ah, I meant using the callback and having lang as a dep though. I don't think we need to pass `onChange` as a dep...

Reply:
Why not? Why can we underspecify `onChange`? To my understanding it closes over it so we won't bust given it changes?

Reply:
I'm just trying to understand why you would need to specify it. Yes, it's a prop and "potentially" it might change but I never saw a case when changing a callback function on purpose would cause the UI to rerender.
It's just confusing for me as a mental model.

We can add it though if you prefer so.

Reply:
I just thought one also has to specify things on `props` as otherwise things won't invalidate the memoization leading to bugs.